File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> mp;
        int numsLen = nums.size();
        
        for(int i = 0; i < numsLen; i++){
            mp[nums[i]]++;
        }
        
        for (auto it = mp.begin(); it != mp.end(); it++) {
          if(it->second > numsLen/2){
              return it->first;
          }
        }
       return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
import java.util.*;

class Solution {
    public int majorityElement(int[] nums) {
        int maj = nums.length / 2;
        Map<Integer, Integer> mpp = new HashMap<>();

        for (int num : nums) {
            mpp.put(num, mpp.getOrDefault(num, 0) + 1);
            if (mpp.get(num) > maj) {
                return num;
            }
        }

        // This line is not strictly needed since the problem guarantees a majority element
        return -1;
    }
}
